megatron/model/transformer.py

```diff
@@ -365,36 +370,47 @@ def forward(self, q, k, v):
         """
 
         assert all((i.dtype in [torch.float16, torch.bfloat16] for i in (q,k,v)))
-        assert all((i.is_cuda for i in (q,k,v)))
+        if get_accelerator().device_name() == 'cuda':
```

Comment:
This could be generalized using the `on_accelerator()` [API](https://github.com/microsoft/DeepSpeed/blob/fb9aebbf254d6d4535e42df75cb08b7c1dbd03f2/accelerator/abstract_accelerator.py#L225)

Reply:
sure, let me prepare another PR to generalize this.